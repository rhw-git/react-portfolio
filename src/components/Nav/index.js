import React from 'react';

import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';

function NavigationBar() {
  return (
    <header>
      <Navbar
        collapseOnSelect
        expand="lg"
        bg="dark"
        variant="dark"
        className="px-4"
      >
        <Navbar.Brand href="/">Portfolio</Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse
          className="justify-content-stretch"
          id="responsive-navbar-nav"
        >
          <Nav className="mr-auto"></Nav>
          <Nav>
            <Nav.Link data-testid="about" href="#about">
              About Me
            </Nav.Link>
            <Nav.Link data-testid="portfolio" href="#portfolio">
              Portfolio
            </Nav.Link>
            <Nav.Link data-testid="contact" href="#contact">
              Contact
            </Nav.Link>
            <Nav.Link data-testid="resume" href="#resume">
              Resume
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </header>
  );
}

export default NavigationBar;
