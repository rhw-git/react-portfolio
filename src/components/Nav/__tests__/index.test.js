// import all library and package
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import component
import Nav from '..';
// delete the component created for testing
afterEach(cleanup);
//---------------tests-----------------//
describe('Nav component', () => {
  // renderts Nav test
  it('render', () => {
    render(<Nav />);
  });
  // arrange
  const { asFragment } = render(<Nav />);
  // assert
  expect(asFragment()).toMatchSnapshot();
});

describe('link visibility', () => {
  it('inserts text into links', () => {
    // arrange
    const { getByTestId } = render(<Nav />);
    // assert
    expect(getByTestId('about')).toHaveTextContent('About Me');
    expect(getByTestId('portfolio')).toHaveTextContent('Portfolio');
    expect(getByTestId('contact')).toHaveTextContent('Contact');
    expect(getByTestId('resume')).toHaveTextContent('Resume');
  });
});
