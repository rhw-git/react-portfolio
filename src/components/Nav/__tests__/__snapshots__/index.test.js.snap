// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="px-4 navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Portfolio
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-stretch navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="mr-auto navbar-nav"
        />
        <div
          class="navbar-nav"
        >
          <a
            class="nav-link"
            data-rb-event-key="#about"
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
          <a
            class="nav-link"
            data-rb-event-key="#portfolio"
            data-testid="portfolio"
            href="#portfolio"
          >
            Portfolio
          </a>
          <a
            class="nav-link"
            data-rb-event-key="#contact"
            data-testid="contact"
            href="#contact"
          >
            Contact
          </a>
          <a
            class="nav-link"
            data-rb-event-key="#resume"
            data-testid="resume"
            href="#resume"
          >
            Resume
          </a>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
